src/test/java/org/klodnicki/service/AccountServiceTest.java

```diff
@@ -0,0 +1,457 @@
+        String lastName = "test";
+        String pesel = "l";
+        String phoneNumber = "2133";
+        String email = "test@test.pl";
+        String address = "test";
+
+        // when - wykonywana akcja, którą testuję
+        //then - porównanie wyniku, który oczekuję do wyniku, który wyszedł
+        // Sprawdzić czy rzucamy IllegalArgumentExcpetion
+        Exception exception = assertThrows(IllegalArgumentException.class, () ->
+                accountService.create(firstName, secondName, lastName, pesel, phoneNumber, email, address));
+
+        // Sprawdzić czy wiadomość jest poprawna
+        assertEquals("Pesel must have 11 digits.", exception.getMessage());
+    }
+
+    @Test
+    public void create_ShouldThrowIllegalArgumentException_WhenPhoneNumberAndEmailAreEmpty() {
+        accountRepository.removeAll();
+        String firstName = "test";
+        String secondName = "test";
+        String lastName = "test";
+        String pesel = "11111111111";
+        String phoneNumber = "";
+        String email = "";
+        String address = "test";
+
+        Exception exception = assertThrows(IllegalArgumentException.class, () -> accountService.create(firstName, secondName, lastName, pesel,
+                phoneNumber, email, address));
+
+        assertEquals("Phone number or email address must have a value.", exception.getMessage());
+    }
+
+    @Test
+    public void create_ShouldCreateAccount_GivenCorrectArguments() {
+        //given
+        accountRepository.removeAll();
+        String firstName = "test";
+        String secondName = "test";
+        String lastName = "test";
+        String pesel = "11111111111";
+        String phoneNumber = "123";
+        String email = "test@test.pl";
+        String address = "test";
+        Long expected = accountRepository.countAccounts() + 1;
+
+        //when
+        // mock here, providing account in db
+        accountService.create(firstName, secondName, lastName, pesel, phoneNumber, email, address);
+
+        //then:
+        Long actual = accountRepository.countAccounts();
+        assertEquals(expected, actual);
+        // Sprawdzenie czy na bazie danych pojawiło się nowe konto
+    }
+
+    @Test
+    public void findAccountByFirstNameAndLastNameAndPesel_ShouldThrowNotFoundInDatabaseException_WhenNoResultIsFoundInDatabase() {
+        //given - dane, które podaję
+        accountRepository.removeAll();
+        String firstName = "";
+        String lastName = "";
+        String pesel = "";
+
+        //when - wykonywana akcja, którą testuję
+        //then - porównanie wyniku oczekiwany do tego co wyszedl
+        assertThrows(NotFoundInDatabaseException.class, () -> accountService.findAccountByFirstNameAndLastNameAndPesel
+                (firstName, lastName, pesel));
+    }
+
+    @Test
+    public void findAccountByFirstNameAndLastNameAndPesel_ShouldNotThrowNotFoundInDatabaseException_WhenResultIsFoundInDatabase() throws NotFoundInDatabaseException {
```

Comment:
Test name is slightly inaccurate. "ShouldNotThrow" would be better to change for something its really testing. Look at the "then" block in code - there is assertEquals, so something is compared rather than check if it's not threw.

Reply:
oh yes! you are absolutely right! I think I must did it in the middle of the night ;)